### PR TITLE
doc: add note on handling of spaces in CommandLineParser

### PR DESCRIPTION
**Pull request by YusukeKameda**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake
Added note that this class will not work properly if tabs and other whitespace characters are included in the key.
The support of whitespace characters by istringstream, etc. is on hold because the future of this class is not clear compared to implementations in Python and other languages.

### CODE REVIEW

File: modules/core/include/opencv2/core/utility.hpp [hunk 1]
```diff
@@ -773,7 +773,7 @@ The sample below demonstrates how to use CommandLineParser:
 The keys parameter is a string containing several blocks, each one is enclosed in curly braces and
 describes one argument. Each argument contains three parts separated by the `|` symbol:
 
--# argument names is a space-separated list of option synonyms (to mark argument as positional, prefix it with the `@` symbol)
+-# argument names is a list of option synonyms separated by ASCII space characters (0x20) (to mark argument as positional, prefix it with the `@` symbol)
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core/utility.hpp:776]: It's too strong statement, as there is no `0x20` constant in the code. The value comes from here: https://github.com/opencv/opencv/blob/0b6c9a2123a8ff1fc0d9e340c8d0132dac3edb6b/modules/core/src/command_line_parser.cpp#L46
It's just space `' '` symbol in source code. I propose to re-write it more generic:  "argument names is a list of option synonyms separated by standard space characters".

Comment (YusukeKameda) [reply modules/core/include/opencv2/core/utility.hpp]: Thank you for pointing this out. I will revise as you suggested.

**Review by YusukeKameda — COMMENTED** [review]
ASCII 0x20 -> standard space ' '

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**